File: PLP_Java/src/Aula03_ListaDeExercicio/Q03_ContaBancaria.java
package Aula03_ListaDeExercicio;

public class Q03_ContaBancaria {
    // Private attributes
    private double saldo;
    private String titular;

    // Constructor
    public Q03_ContaBancaria(String titular, double saldoInicial) {
        this.saldo = saldoInicial;
        this.titular = titular;
    }

    // Method to deposit money into the account
    public void depositar(double valor) {
        if (valor > 0) {
            this.saldo += valor;
        } else {
            System.out.println("O valor do depósito deve ser positivo.");
        }
    }

    // Method to withdraw money from the account
    public void sacar(double valor) {
        if (valor > 0 && valor <= this.saldo) {
            this.saldo -= valor;
        } else {
            System.out.println("Saldo insuficiente ou valor inválido.");
        }
    }

    // Method to check the current balance
    public double consultarSaldo() {
        return this.saldo;
    }

    // Main method to test the class
    public static void main(String[] args) {
        Q03_ContaBancaria conta = new Q03_ContaBancaria("João", 1000);
        System.out.println("Saldo inicial: R$" + conta.consultarSaldo());
        conta.depositar(500);
        System.out.println("Saldo após depósito: R$" + conta.consultarSaldo());
        conta.sacar(300);
        System.out.println("Saldo após saque: R$" + conta.consultarSaldo());
    }
}

File: PLP_python/Aula03_ListaDeExercicios(15)/Q03_ContaBancaria.py
class ContaBancaria:
    """
    A class used to represent a Bank Account.
    """

    def __init__(self, titular, saldo_inicial=0):
        """
        The constructor for the ContaBancaria class.
        """
        self.__saldo = saldo_inicial  # Private attribute
        self.titular = titular

    def depositar(self, valor):
        """
        Deposits a given amount into the account.
        """
        if valor > 0:
            self.__saldo += valor
        else:
            print("O valor do depósito deve ser positivo.")

    def sacar(self, valor):
        """
        Withdraws a given amount from the account, if sufficient funds are available.
        """
        if 0 < valor <= self.__saldo:
            self.__saldo -= valor
        else:
            print("Saldo insuficiente ou valor inválido.")

    def consultar_saldo(self):
        """
        Returns the current balance of the account.
        """
        return self.__saldo


# Example usage
conta = ContaBancaria("João", 1000)
print("Saldo inicial: R$", conta.consultar_saldo())
conta.depositar(500)
print("Saldo após depósito: R$", conta.consultar_saldo())
conta.sacar(300)
print("Saldo após saque: R$", conta.consultar_saldo())
